frontend/packages/knative-plugin/src/hooks/useEventSourceStatus.ts

```diff
@@ -0,0 +1,92 @@
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+import { K8sKind, K8sResourceKind } from '@console/internal/module/k8s';
+import { checkAccess } from '@console/internal/components/utils';
+import { useK8sGet } from '@console/internal/components/utils/k8s-get-hook';
+import { getEventSourceMetadata, getKameletMetadata } from '../utils/create-eventsources-utils';
+import { EventSourceMetaData } from '../components/add/import-types';
+import { CamelKameletBindingModel, CamelKameletModel } from '../models';
+import {
+  isDynamicEventSourceKind,
+  useEventSourceModels,
+} from '../utils/fetch-dynamic-eventsources-utils';
+
+export const useEventSourceStatus = (
+  namespace: string,
+  sourceKindProp?: string,
+  kameletName?: string,
+): {
+  sourceFound: boolean;
+  createSourceAccess: boolean;
+  loaded: boolean;
+  normalizedSource: EventSourceMetaData;
+  kamelet: K8sResourceKind;
+} => {
+  const { t } = useTranslation();
+  const { eventSourceModels, loaded: eventSourceModelsLoaded } = useEventSourceModels();
+  const [kamelet, kameletLoaded] = useK8sGet<K8sResourceKind>(
+    CamelKameletModel,
+    kameletName,
+    namespace,
+  );
+  const [createSourceAccess, setCreateSourceAccess] = React.useState<boolean>(true);
+  const [sourceFound, setSourceFound] = React.useState<boolean>(true);
+  const [loaded, setLoaded] = React.useState<boolean>(false);
+  const [normalizedSource, setNormalizedSource] = React.useState<EventSourceMetaData>();
```

Comment:
Not sure if we are going to need all these states. I think We can get away with using `useMemo` for these.